src/game/features/self/InfiniteAmmo.cpp

```diff
@@ -11,17 +11,12 @@ namespace YimMenu::Features
 
 		virtual void OnTick() override
 		{
-			Hash current_weapon{};
-			int max_ammo{};
-
-			WEAPON::GET_CURRENT_PED_WEAPON(Self::PlayerPed, &current_weapon, false, false, false);
-			WEAPON::GET_MAX_AMMO(Self::PlayerPed, &max_ammo, current_weapon);
-
-			auto current_weapon_ammo_type = WEAPON::_GET_CURRENT_PED_WEAPON_AMMO_TYPE(Self::PlayerPed, WEAPON::_GET_PED_WEAPON_OBJECT(Self::PlayerPed, false));
-			auto current_ammo = WEAPON::GET_PED_AMMO_BY_TYPE(Self::PlayerPed, current_weapon_ammo_type);
+			WEAPON::SET_PED_INFINITE_AMMO(Self::PlayerPed, true, 0);
```

Comment:
Not sure whether using natives here would be desirable, since they tend to do some extra stuff I'd rather avoid.

Reply:
Figures.